pkg/router/template/plugin_test.go

```diff
@@ -222,25 +220,25 @@ func (r *TestRouter) FilterNamespaces(namespaces sets.String) {
 	for k := range r.ServiceUnits {
 		// TODO: the id of a service unit should be defined inside this class, not passed in from the outside
 		//   remove the leak of the abstraction when we refactor this code
-		ns := strings.SplitN(k, "/", 2)[0]
+		ns, _ := getPartsFromEndpointsKey(k)
 		if namespaces.Has(ns) {
 			continue
 		}
 		delete(r.ServiceUnits, k)
 	}
 
 	for k := range r.State {
-		ns := strings.SplitN(k, "-", 2)[0]
+		ns, _ := getPartsFromRouteKey(k)
 		if namespaces.Has(ns) {
 			continue
 		}
 		delete(r.State, k)
 	}
 }
 
-// routeKey create an identifier for the route consisting of host-path
-func (r *TestRouter) routeKey(route *routeapi.Route) string {
-	return route.Spec.Host + "-" + route.Spec.Path
+// getKey create an identifier for the route consisting of host-path
+func getKey(route *routeapi.Route) string {
+	return route.Spec.Host + routeKeySeparator + route.Spec.Path
```

Comment:
Any idea why this doesn't use sprintf?  Or why the other ones don't do it this way?

Reply:
And should this call routeKeyFromParts?

Reply:
fmt.Sprintf and ' + ' are used in several places and looks like we don't care which one we use.
Yes, we could call routeKeyFromParts